tst/regression/CMakeLists.txt

```diff
@@ -22,18 +22,35 @@
 
 # 2. the correct arguments to pass to the 'run_test.py' script must be specfied
 #   * This arguments must be added to the CMAKE variable TEST_ARGS as shown below
+#   * The number of mpi processors that should be used when testing the mpi implementation 
 #
 # Note each new test must append exactly 1 item to each of the CMAKE variables
 
+function(record_driver arg)
+    list(LENGTH arg len_list)
+    math(EXPR list_end "${len_list} - 1")
+    foreach(ind RANGE ${list_end})
+      list(GET arg ${ind} arg2)
+      if("${arg2}" STREQUAL "--driver")
+        MATH(EXPR ind "${ind}+1")
+        list(GET arg ${ind} driver)
+        get_filename_component(driver ${driver} NAME)
+        set_property(GLOBAL APPEND PROPERTY DRIVERS_USED_IN_TESTS "${driver}" )
+      endif()
+    endforeach()
+endfunction()
```

Comment:
As someone with very poor cmake skills, I think I would benefit from an explanatory comment that describes exactly what this is doing. But it's not a big deal either way.

Reply:
Good point!